MLE-13687 Able to use onError in plan builder

rjrudin opened the pull request:
Regenerated the code so that `onError` appears on `ModifyPlan` instead of `PlanBuilder`.

BillFarber reviewed (commented):
[2024-04-09T12:47:38.907Z] ErrorListenerInputOutputEndpointTest > testInputOutputCallerWithStop() FAILED
[2024-04-09T12:47:38.907Z]     org.opentest4j.AssertionFailedError at ErrorListenerInputOutputEndpointTest.java:219

[2024-04-09T12:48:05.458Z] OpticPatchTest > replaceInsertChild() FAILED
[2024-04-09T12:48:05.458Z]     org.opentest4j.AssertionFailedError at OpticPatchTest.java:132

rjrudin commented:
> [2024-04-09T12:47:38.907Z] ErrorListenerInputOutputEndpointTest > testInputOutputCallerWithStop() FAILED [2024-04-09T12:47:38.907Z] org.opentest4j.AssertionFailedError at ErrorListenerInputOutputEndpointTest.java:219
> 
> [2024-04-09T12:48:05.458Z] OpticPatchTest > replaceInsertChild() FAILED [2024-04-09T12:48:05.458Z] org.opentest4j.AssertionFailedError at OpticPatchTest.java:132

The first is one of a handful of intermittent Bulk DS failures. 

The second is interesting - it's not related to this PR. I believe a fix was made to 11-nightly because when I added this test yesterday, the expected behavior didn't seem correct. But now it does, and so I updated the test to match the behavior of the server. That behavior is - `replaceInsertChild` now replaces the existing element with text. It was not doing that yesterday, but it is today.
